src/App/UI/ConversationViewUI.java

```diff
@@ -119,7 +135,13 @@ public void actionlistener() {
             mainUI.placeFrameInCoordinates(frame.getX(),frame.getY());
             mainUI.setVisible(true);
         });
-
+        searchButton.addActionListener(e -> {
+            String chatId = MessagesRepository.mr().getChatId(receiverUsername);
+            user.sendMessagesToRemoteServer(chatId);
+            mainUI.closeChat();
```

Comment:
we should call peer method searchForKeywords here and we should not close the chat